src/opencl_iwork_fmt_plug.c

```diff
@@ -0,0 +1,401 @@
+#else
+
+#include <string.h>
+#include "aes.h"
+#include "hmac_sha.h"
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#include "arch.h"
+#include "formats.h"
+#include "common.h"
+#include "stdint.h"
+#include "iwork_common.h"
+#include "options.h"
+#include "jumbo.h"
+#include "sha2.h"
+#include "common-opencl.h"
+#include "misc.h"
+#define OUTLEN (16)
+#include "opencl_pbkdf2_hmac_sha1.h"
+
+#define FORMAT_LABEL            "iwork-opencl"
+#define FORMAT_NAME             "Apple iWork '09 / '13 / '14"
+#define FORMAT_TAG              "$iwork$"
+#define TAG_LENGTH              (sizeof(FORMAT_TAG) - 1)
+#define OCL_ALGORITHM_NAME      "PBKDF2-SHA1 OpenCL"
+#define CPU_ALGORITHM_NAME      " AES"
+#define ALGORITHM_NAME          OCL_ALGORITHM_NAME CPU_ALGORITHM_NAME
+#define BENCHMARK_COMMENT       ""
+#define BENCHMARK_LENGTH        -1
+#define MIN_KEYS_PER_CRYPT      1
+#define MAX_KEYS_PER_CRYPT      1
+
+#define BINARY_SIZE             0
+#define BINARY_ALIGN            MEM_ALIGN_WORD
+#define PLAINTEXT_LENGTH        64
+#define SALT_SIZE               sizeof(*fctx)
+#define SALT_ALIGN              MEM_ALIGN_WORD
+
+/* This handles all widths */
+#define GETPOS(i, index)        (((index) % ocl_v_width) * 4 + ((i) & ~3U) * ocl_v_width + (((i) & 3) ^ 3) + ((index) / ocl_v_width) * 64 * ocl_v_width)
+
+static int *cracked;
+static int any_cracked;
+
+static const int KEY_CHECKSUM_BYTES = 4;
+
+static iwork_common_custom_salt *fctx;
+
+static struct fmt_tests iwork_tests[] = {
+	{"$iwork$1$2$1$100000$d77ce46a68697e08b76ac91de9117541$e7b72b2848dc27efed883963b00b1ac7$e794144cd2f04bd50e23957b30affb2898554a99a3accb7506c17132654e09c04bbeff45dc4f8a8a1db5fd1592f699eeff2f9a8c31b503e9631a25a344b517f7" ,"12345678"},
+	{FORMAT_TAG "1$2$1$100000$c773f06bcd580e4afa35618a7d0bee39$8b241504af92416f226d0eea4bf26443$18358e736a0401061f2dca103fceb29e88606d3ec80d09841360cbb8b9dc1d2908c270d3ff4c05cf7a46591e02ff3c9d75f4582f631721a3257dc087f98f523e", "password"},
+	// iWork '09 Keynote file
+	{"$iwork$2$1$1$4000$736f6d6553616c74$a9d975f8b3e1bf0c388944b457127df4$09eb5d093584376001d4c94e9d0a41eb8a2993132849c5aed8e56e7bd0e8ed50ba38aced793e3480675990c828c01d25fe245cc6aa603c6cb1a0425988f1d3dc", "openwall"},
+	{NULL}
+};
+
+static size_t key_buf_size;
+static unsigned int *inbuffer;
+static pbkdf2_out *output;
+static pbkdf2_salt currentsalt;
+static cl_mem mem_in, mem_out, mem_salt, mem_state;
+static size_t key_buf_size;
+static int new_keys;
+static struct fmt_main *self;
+
+static cl_kernel pbkdf2_init, pbkdf2_loop, pbkdf2_final;
+
+#define cracked_size (sizeof(*cracked) * global_work_size * ocl_v_width)
+
+/*
+ * HASH_LOOPS is ideally made by factors of (iteration count - 1) and should
+ * be chosen for a kernel duration of not more than 200 ms
+ */
+#define HASH_LOOPS              (3 * 251)
+#define ITERATIONS              181474 /* Just for auto tune */
+#define LOOP_COUNT              (((currentsalt.iterations - 1 + HASH_LOOPS - 1)) / HASH_LOOPS)
```

Comment:
HASH_LOOPS should better be `(3 * 271)` for being optimized for 100,000 iterations. And ITERATIONS should be `100000` for better auto-tune.